src/jit/codegencommon.cpp

```diff
@@ -10317,10 +10333,23 @@ void CodeGen::genFuncletProlog(BasicBlock* block)
 
     compiler->unwindBegProlog();
 
+    inst_RV(INS_push, REG_FPBASE, TYP_REF);
+    compiler->unwindPush(REG_FPBASE);
+
     // TODO Save callee-saved registers
 
     // This is the end of the OS-reported prolog for purposes of unwinding
     compiler->unwindEndProlog();
+
+    // Restore frame pointer
+    // TODO We will need changes here when we introduce PSPSym
+    inst_RV_RV(INS_mov, REG_FPBASE, REG_ARG_0);
```

Comment:
Here are register dumps.

on breakpoint `call edx`
```
eax            0xf30149bc       -218019396
ecx            0xffffc434       -15308
edx            0xf25e24c2       -228711230
ebx            0xffffd2a8       -11608
esp            0xffffb924       0xffffb924
ebp            0xffffcc38       0xffffcc38
esi            0xf301237c       -218029188
edi            0xffffcc2c       -13268
eip            0xf719b23b       0xf719b23b <CallEHFunclet+31>
```
when abortion at frame 8
```
#8  0xf719b23d in CallEHFunclet () at ehhelpers.S:156
156     ehhelpers.S: No such file or directory.
(gdb) i r
eax            0x0      0
ecx            0x8      8
edx            0x6      6
ebx            0xf719b23d       -149310915
esp            0xffffb920       0xffffb920
ebp            0xffffcc38       0xffffcc38
esi            0x0      0
edi            0xf7a1c911       -140392175
eip            0xf719b23d       0xf719b23d <CallEHFunclet+33>
```

Like you said, ESPs differ.